src/FileReader/YAMLFileReader.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace TomPHP\ContainerConfigurator\FileReader;
+
+use Assert\Assertion;
+use Symfony\Component\Yaml;
+use TomPHP\ContainerConfigurator\Exception\InvalidConfigException;
+
+final class YAMLFileReader implements FileReader
+{
+    public function read($filename)
+    {
+        Assertion::file($filename);
+
```

Comment:
We should assert the dependency is available here. Something like:

```php
if (!class_exists(Symfony\Component\Yaml\Yaml::class)) {
    throw MissingDependency::fromPackageName('symfony/yaml');
}
```

And let's make the exception message explain that the dependency should be installed for a nice user experience.